packages/core/src/transport/httpRequest.spec.ts

```diff
@@ -1,101 +1,105 @@
 /* eslint-disable @typescript-eslint/unbound-method */
-import sinon from 'sinon'
-import { stubEndpointBuilder } from '../../test/specHelper'
+import { stubEndpointBuilder, interceptRequests } from '../../test/specHelper'
+import type { Request } from '../../test/specHelper'
 import type { EndpointBuilder } from '../domain/configuration'
 import { createEndpointBuilder } from '../domain/configuration'
 import { createHttpRequest } from './httpRequest'
 import type { HttpRequest } from './httpRequest'
 
 describe('httpRequest', () => {
   const BATCH_BYTES_LIMIT = 100
-  let server: sinon.SinonFakeServer
+  let requests: Request[]
   let endpointBuilder: EndpointBuilder
   let request: HttpRequest
   const ENDPOINT_URL = 'http://my.website'
 
   beforeEach(() => {
-    server = sinon.fakeServer.create()
+    requests = interceptRequests()
     endpointBuilder = stubEndpointBuilder(ENDPOINT_URL)
     request = createHttpRequest(endpointBuilder, BATCH_BYTES_LIMIT)
   })
 
-  afterEach(() => {
-    server.restore()
-  })
-
   it('should use xhr when sendBeacon is not defined', () => {
+    const originalSendBeacon = navigator.sendBeacon.bind(navigator)
+    ;(navigator.sendBeacon as any) = false
+
     request.send('{"foo":"bar1"}\n{"foo":"bar2"}', 10)
 
-    expect(server.requests.length).toEqual(1)
-    expect(server.requests[0].url).toContain(ENDPOINT_URL)
-    expect(server.requests[0].requestBody).toEqual('{"foo":"bar1"}\n{"foo":"bar2"}')
+    expect(requests.length).toEqual(1)
+    expect(requests[0].type).toBe('xhr')
+    expect(requests[0].url).toContain(ENDPOINT_URL)
+    expect(requests[0].body).toEqual('{"foo":"bar1"}\n{"foo":"bar2"}')
+    navigator.sendBeacon = originalSendBeacon
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** restore the state in an `afterEach` callback